Modelica/Media/R134a.mo

```diff
@@ -326,19 +326,26 @@ rho = Medium.density(setState_phX(p, h, fill(0, Medium.nX)));
 
     algorithm
       R_s := R134aData.data.R_s;
-      f := f_R134a(d=d, T=T);
-      state.p := d*R_s*T*f.delta*f.fdelta;
-      state.h := R_s*T*(f.tau*f.ftau + f.delta*f.fdelta);
-      sat.psat := state.p;
-      state.T := T;
-      state.d := d;
-      sat.Tsat := state.T;
+      sat := setSat_T(T);
       dl := bubbleDensity(sat);
       dv := dewDensity(sat);
-      state.phase := if ((state.d < dl) or (state.d > dv) or (state.p >
-        R134aData.data.FPCRIT)) then 1 else 2;
-
+      if d < dl and d > dv and T < R134aData.data.FTCRIT then
+        f := Modelica.Media.Common.HelmholtzDerivs(
+          d=d, T=T, R_s=R_s, delta=0, tau=0, f=0, fdelta=0,
+          fdeltadelta=0, ftau=0, ftautau=0, fdeltatau=0);
+        state.p := saturationPressure(T);
+        state.h := 1/(dl/dv + 1)*(dewEnthalpy(sat) - bubbleEnthalpy(sat)) + bubbleEnthalpy(sat);
```

Comment:
I'm not sure this line is correct, first and foremost because `state.h` should obviously depend on `d`.

AFAIK, specific volume v (not density) is linear with enthalpy, hence:
```
v = vl*(hv-h)/(hv-hl) + vv*(h-hl)/(hv-hl)
v*(hv-hl) = vl*hv - vl*h + vv*h - vv*hl
v*(hv-hl) - vl*hv + vv*hl = (vv-vl)*h
h = (v*(hv-hl) - vl*hv + vv*hl)/(vv-vl)
h = (1/d*(hv-hl) - 1/dl*hv + 1/dv*hl)/(1/dv-1/dl)
h = (dl*dv/d*(hv-hl) - dv*hv + dl*hl)/(dl-dv)
```
when d = dl you get 
```
h = (dv*(hv-hl) - dv*hv + dl*hl)/(dl-dv) = (hv*(dv-dv) + hl*(dl-dv))/(dl-dv) = hl
```
when d = dv you get 
```
h = (dl*(hv-hl) - dv*hv + dl*hl)/(dl-dv) = (hv*(dl-dv) + hl*(dl-dl)/(dl-dv) = hv
```
Furthermore, the right-hand side of
```
h = (dl*dv/d*(hv-hl) - dv*hv + dl*hl)/(dl-dv)
```
can be divided by dl at both numerator and denominator, yielding
```
h = (dv/d*(hv-hl) -dv/dl*hv + hl)/(1-dv/dl)
h = (dv*dl*(hv-hl) - dv*d*hv + d*dl*hl)/((d*dl)*(1-dv/dl))
h = (hv*(dv*dl-dv*d)+hl*(d*dl-dv*dl)/(d*(dl-dv))
h = (hv*dv*(dl-d) + hl*dl*(d-dv))/(d*(dl-dv))
h = (hv*dv*(dl/d-1) + hl*dl*(1-dv/d))/(dl-dv)
```
but I guess the last equation is not simpler than 
```
h = (dl*dv/d*(hv-hl) - dv*hv + dl*hl)/(dl-dv)
```
I couldn't find any simpler representation.